src/main/java/se/sundsvall/smloader/integration/util/XPathUtil.java

```diff
@@ -0,0 +1,146 @@
+package se.sundsvall.smloader.integration.util;
+
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.springframework.core.annotation.AnnotationUtils;
+import se.sundsvall.smloader.integration.util.annotation.XPath;
+import us.codecraft.xsoup.Xsoup;
+
+import java.lang.reflect.Field;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+import static java.lang.reflect.Modifier.isAbstract;
+import static java.lang.reflect.Modifier.isInterface;
+import static java.util.Optional.ofNullable;
+import static java.util.function.Predicate.not;
+import static org.jsoup.Jsoup.parse;
+import static org.jsoup.parser.Parser.xmlParser;
+
+public final class XPathUtil {
```

Comment:
This class contains both the annotaion processing part and some xpath evaluation stuff, from what I can see?

Maybe introduce separation of concerns here?

I.e. move the the annotation-processing stuff to a class named "XPathAnnotationProcessor" (or something like that) and put it in the same package as "Xpath" (or a impl subpackage)?

The xpath util stuff can be kept in this class.

Thoughts?

Reply:
That seems like a good idea I think. I stole this from another service and the main focus was to make it work. I think this is useful for other services as well and should be moved to dept44 later.